src/System.Transactions/src/System/Transactions/TransactionsEtwProvider.cs

```diff
@@ -203,14 +203,17 @@ internal void TransactionCreated(Transaction transaction, string type)
 
             if (IsEnabled(EventLevel.Informational, ALL_KEYWORDS))
             {
-                TransactionCreated(
-                    transaction.TransactionTraceId.TransactionIdentifier, type);
+                if (transaction != null && transaction.TransactionTraceId != null)
+                    TransactionCreated(transaction.TransactionTraceId.TransactionIdentifier, type);
+                else
+                    TransactionCreated(string.Empty, type);
             }
         }
 
         [Event(TRANSACTION_CREATED_EVENTID, Keywords = Keywords.TraceLtm, Level = EventLevel.Informational, Task = Tasks.Transaction, Opcode = Opcodes.Create, Message = "Transaction Created. ID is {0}, type is {1}")]
         private void TransactionCreated(string transactionIdentifier, string type)
         {
+            SetActivityId(transactionIdentifier);
```

Comment:
>transactionIdentifier [](start = 26, length = 21)

On further reflection, the same Guid is used for all "local" transactions. The InternalTransaction object's HashCode is what differentiates the transactions. The TransactionTraceId.TransactionIdentifier string looks like "<guid>:<hashcode>". And for local transactions, the <guid> portion is always the same for the System.Transactions.dll in a given process. So using this Guid portion as the ActivityId is not useful.